src/simulators/statevector/chunk/chunk_manager.hpp

```diff
@@ -43,12 +43,15 @@ class ChunkManager
   int num_qubits_;             //number of global qubits
 
   uint_t num_chunks_;          //number of chunks on this process
+  uint_t chunk_index_;         //global chunk index for the first chunk
```

Comment:
I'm confused why previous builds work well without `chunk_index_`. I guess that this is necessary in MPI execution for each process to know what is the first chunk in the process.

Reply:
This is handled in `QubitVectorThrust` previously to pass this value to kernel functions. But in this PR I moved all the kernel functions in `ChunkContainer` that requires this parameter.